### PR TITLE
fix(1095): Add command arguments field [1]

### PR DESCRIPTION
**Pull request by tkyi**
## Context
It would be nice if users could add optional arguments to their command usage.

## Objective
This PR adds an optional field "arguments" to commands.

## Related links
Related to https://github.com/screwdriver-cd/screwdriver/issues/1095

### CODE REVIEW

File: config/command.js [hunk 1]
```diff
@@ -45,6 +45,13 @@ const COMMAND_DESCRIPTION = Joi
     .description('Description of the Command')
     .example('Installs npm modules');
 
+const COMMAND_ARGUMENTS = Joi
+    .string()
+    .max(256)
```
Comment (tk3fftk) [inline config/command.js]: Is 256 enough?  I imagine the arguments view can be like CLI command.
```
Usage: sd-cmd exec foo/bar@1.2.3 <OPTION> <TARGET>
Options:
        --config               some config file
        --debug               debug mode (default "false")
        --host                  super host
        --log-level           Set the logging level ("debug"|"info"|"warn"|"error"|"fatal") (default "info")
Target:
...
```

File: test/data/command.create.yaml [hunk 2]
```diff
@@ -4,7 +4,14 @@ name: bar
 version: '1.0'
 description: |
   Lorem ipsum dolor sit amet.
-arguments: '-d <domain> -h <host>'
+arguments: '<OPTION> <TARGET>
```
Comment (tk3fftk) [inline test/data/command.create.yaml]: Do we need `|` to keep new lines?

Comment (tkyi) [reply test/data/command.create.yaml]: Newlines are not respected by the UI, so I always just assumed we used it for easier readability in the yaml file.

**Comment by tkyi** [comment]
@d2lam `usage`? But I'm worried people will think they need to write the whole `sd-cmd exec foo/bar@1.2.3`

**Comment by tkyi** [comment]
**Update 7/16/2018:** 
Changed from `arguments` to `usage`. If `usage` field is set, it will override the usage field in the UI. Otherwise, the default will be used (as before).

File: test/data/command.create.yaml (hunk 2 above)
Comment (tkyi) [reply test/data/command.create.yaml]: Sorry I just realized you meant I should add `|`
Yes, you need to add `|` for new lines.

**Pull request merged**